src/main/java/org/broadinstitute/consent/http/resources/UserResource.java

```diff
@@ -53,6 +57,52 @@ public UserResource(UserService userService, LibraryCardService libraryCardServi
         this.gson = new Gson();
     }
 
+    @GET
+    @Produces("application/json")
+    @Path("/role/{roleName}")
+    @RolesAllowed({ADMIN, SIGNINGOFFICIAL})
+    public Response getUsers(@Auth AuthUser authUser, @PathParam("roleName") String roleName) {
+        try {
+            User user = userService.findUserByEmail(authUser.getName());
+            if (Objects.nonNull(roleName)) {
+                boolean valid = EnumSet.allOf(UserRoles.class)
+                  .stream()
+                  .map(UserRoles::getRoleName)
+                  .map(String::toLowerCase)
+                  .anyMatch(roleName::equalsIgnoreCase);
+                if (valid) {
+                    //if the roleName is SO and the user does not have that role throw an exception
+                    if (roleName.equals(UserRoles.SIGNINGOFFICIAL.getRoleName())) {
+                        if (!user.hasUserRole(UserRoles.SIGNINGOFFICIAL)) {
+                            throw new NotFoundException("User: " + user.getDisplayName() + ", " + " does not have Signing Official role.");
```

Comment:
nit: remove all the messaging here and in other places. I'm torn on whether we should throw a not found here too. Logically, it should be a 403 and then we let the dispatch handler in `Resource` translate 403's to 404's, but I don't think we're doing that right now. 

Additionally, we should be validating the user role prior to all of these if blocks ... just after `if(valid) {...` so that we are prepared for any case where a user tries to see a role they shouldn't have.

Reply:
In the swagger, this case is marked as a 400 error, not a 404.

Reply:
"Additionally, we should be validating the user role prior to all of these if blocks" - I'm not sure I understand what you mean by this... because these if block are validating the role name. And do you want me to remove the exception message completely with no replacement or ? I had originally had the error as a 403 and then after our convo yesterday about how that is leaking unnecessary info, I changed it to 400 bad request, and then I changed it to 404 not found, what do you think, leave it as 404?

Reply:
> "Additionally, we should be validating the user role prior to all of these if blocks"

What I mean here is that this code looks at role names one by one looking for only the ones we care about now. Validating that the user has the role they are asking for is a singular action that should be valid for all users asking for any role. After `isValid`, we should ensure that the user has the role they are asking for and immediately throw an error in that case (failing fast) instead of looking at roles one by one (failing slowly).

Reply:
> what do you think, leave it as 404

Yes, let's leave it like that for now. We can have a team conversation about how we translate authorization errors to 403s or 404s and make a decision at another time.

Reply:
Okay that makes sense thank you!